kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/pages/MainPage.java

```diff
@@ -2,15 +2,58 @@
 
 import com.codeborne.selenide.Condition;
 import io.qameta.allure.Step;
+import lombok.SneakyThrows;
 import org.openqa.selenium.By;
 
-import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.*;
 
 public class MainPage {
 
-    @Step
-    public void shouldBeOnPage(){
-        $(By.xpath("//*[contains(text(),'Loading')]")).shouldBe(Condition.disappear);
-        $(By.xpath("//h5[text()='Clusters']")).shouldBe(Condition.visible);
+  private static final long TIMEOUT = 25000;
+
+  @Step
+  public MainPage shouldBeOnPage() {
+    $(By.xpath("//*[contains(text(),'Loading')]")).shouldBe(Condition.disappear);
+    $(By.xpath("//h5[text()='Clusters']")).shouldBe(Condition.visible);
+    return this;
+  }
+
+
+  private void refreshUntil(By by){
+    int i =0;
+    do
+    {
+      refresh();
+      i++;
+      sleep(2000);
+    } while(getElements(by).size()<1 && i!=20);
+    $(by).shouldBe(Condition.visible);
```

Comment:
Doesn't `$(by).waitUntil` works here instead?

Reply:
for what purpose? We already waited necessary time in a loop above

Reply:
instead of the loop